# ANOVA
ANOVA, or Analysis of Variance, is a statistical method used to determine whether there are significant differences between the means of three or more groups. It helps to analyze whether variations in a dataset are due to random chance or are influenced by the independent variable you're testing.

Doctors want to test a new medication which reduce headache they splits
the particular into 3 conditions [15mg,30mg,45mg]
later on the doctor ask the patient to rate headache between(1-10) Are there any difference using medicine in 3 conditions using alpha - 0.05?
Frequencies are:-
15mg -> [9,8,7,8,8,9,8]
30mg -> [7,6,6,7,8,7,8]
45mg -> [4,3,2,3,4,3,2]

In [ ]:
import pandas as pd
from scipy.stats import f_oneway

#  f_oneway is a function from the SciPy library in Python that performs 
#  a One-Way ANOVA (Analysis of Variance) test. It is used to determine 
#  whether there are significant differences between the means of two or 
#  more independent groups

# Step 1: Input the headache ratings for each dosage
ratings_15mg = [9, 8, 7, 8, 8, 9, 8]
ratings_30mg = [7, 6, 6, 7, 8, 7, 8]
ratings_45mg = [4, 3, 2, 3, 4, 3, 2]

# Step 2: Create a pandas DataFrame (optional, for organization)
data = pd.DataFrame({
    "15mg": ratings_15mg,
    "30mg": ratings_30mg,
    "45mg": ratings_45mg
})
print("Data:\n", data)

# Step 3: Perform ANOVA using scipy.stats.f_oneway
f_statistic, p_value = f_oneway(ratings_15mg, ratings_30mg, ratings_45mg)

# Step 4: Display the results
print("F-Statistic:", f_statistic)
print("P-value:", p_value)

# Step 5: Interpret the results
alpha = 0.05  # Significance level
if p_value < alpha:
    print("Reject the null hypothesis: There is a significant difference between the headache ratings across conditions.")
else:
    print("Fail to reject the null hypothesis: There is no significant difference between the headache ratings across conditions.")

Data:
    15mg  30mg  45mg
0     9     7     4
1     8     6     3
2     7     6     2
3     8     7     3
4     8     8     4
5     9     7     3
6     8     8     2
F-Statistic: 84.63157894736862
P-value: 7.004569693845702e-10
Reject the null hypothesis: There is a significant difference between the headache ratings across conditions.


F-Statistic: The F-statistic of 84.63 is a measure of how much variability exists between the groups (15mg, 30mg, and 45mg dosages) relative to the variability within the groups. In your case, it is calculated as the ratio of between-group variance to within-group variance.